test/helpers/components/edgeconnect/edgeconnect.go

```diff
@@ -0,0 +1,68 @@
+//go:build e2e
```

Comment:
AFAIK, the e2e tests are out of scope for this change

Reply:
Having proper tests from the beginning is a good thing, this includes E2E tests. They also provide an reproducible way beyond "works on my machine" of showing, that the controller works. Or quickly detect it doesn't. So while they were not mentioned in the ticket, I've taken the freedom to add them as I see them as a mandatory part of a new feature.

Reply:
the thing is that we have other stuff that depends on having the base reconciler in the code but at the same time you did some refactoring in the e2e tests, which touches almost all scenarios there.
If the e2e tests were just a drop in thing, no major changes, then I would have no problem with it.
But the current state for the e2e tests and the helpers we have there are veeeeery dynakube/ single CR focused, which needs rethinking/refactoring, and I would prefer to do that rethinking not in this PR.

So if you really want e2e tests to be part of this change, then pls don't refactor the majority of the e2e logic, or we will be here forever, discussing how the e2e helpers should be refactored 😅 

I would prefer to tackle e2e tests in a separate PR, like we always did, furthermore we only add e2e tests when a feature is actually complete, while this PR only introduces a very simple part of the feature, not much to test.

Reply:
Fair point.